[CI] XPackRestIT test {p0=transform/transforms_unattended/Test unattended put and start} failing

reproduces if I slow down the validation call, which seems to happen on CI.

**Build scan:**
https://gradle-enterprise.elastic.co/s/n5k3wukhl7ksy/tests/:x-pack:plugin:yamlRestTest/org.elasticsearch.xpack.test.rest.XPackRestIT/test%20%7Bp0=transform%2Ftransforms_unattended%2FTest%20unattended%20put%20and%20start%7D

**Reproduction line:**
`./gradlew ':x-pack:plugin:yamlRestTest' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=transform/transforms_unattended/Test unattended put and start}" -Dtests.seed=92EAE2D17C809F82 -Dtests.locale=ar-SD -Dtests.timezone=America/Yellowknife -Druntime.java=17`

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.test.rest.XPackRestIT&tests.test=test%20%7Bp0%3Dtransform/transforms_unattended/Test%20unattended%20put%20and%20start%7D

**Failure excerpt:**
```
java.lang.AssertionError: Failure at [transform/transforms_unattended:14]: expected [2xx] status code but api [transform.stop_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Failed to update transform task [transform-unattended] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [transform-unattended] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:329)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:305)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1362)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1447)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1427)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:825)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"status_exception","reason":"Failed to update transform task [transform-unattended] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [transform-unattended] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:329)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:305)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1362)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1447)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1427)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:825)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"status":409}]

  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:510)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:484)
  at jdk.internal.reflect.GeneratedMethodAccessor13.invoke(null:-1)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: java.lang.AssertionError: expected [2xx] status code but api [transform.stop_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Failed to update transform task [transform-unattended] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [transform-unattended] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:329)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:305)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1362)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1447)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1427)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:825)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"status_exception","reason":"Failed to update transform task [transform-unattended] state value should_stop_at_checkpoint from [false] to [false]","stack_trace":"org.elasticsearch.ElasticsearchStatusException: Failed to update transform task [transform-unattended] state value should_stop_at_checkpoint from [false] to [false]\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.buildException(TransportStopTransformAction.java:329)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$waitForStopListener$9(TransportStopTransformAction.java:305)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.finishHim(TransportTasksAction.java:323)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onOperation(TransportTasksAction.java:297)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:277)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleResponse(TransportTasksAction.java:269)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1362)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1447)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1427)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:134)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.support.tasks.TransportTasksAction$1.onFailure(TransportTasksAction.java:114)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$3(TransportStopTransformAction.java:246)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:170)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:164)\n\tat org.elasticsearch.xpack.transform.transforms.TransformTask.setShouldStopAtCheckpoint(TransformTask.java:319)\n\tat org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$taskOperation$4(TransportStopTransformAction.java:238)\n\tat org.elasticsearch.server@8.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:825)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"status":409}]

    at org.junit.Assert.fail(Assert.java:88)
    at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:373)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:499)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:484)
    at jdk.internal.reflect.GeneratedMethodAccessor13.invoke(null:-1)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:568)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
    at java.lang.Thread.run(Thread.java:833)

```

--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexer.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexer.java
@@ -418,20 +418,17 @@ public abstract class AsyncTwoPhaseIndexer<JobPosition, JobStats extends Indexer
 
     private void finishWithSearchFailure(Exception exc) {
         stats.incrementSearchFailures();
-        onFailure(exc);
-        doSaveState(finishAndSetState(), position.get(), this::afterFinishOrFailure);
+        finishWithFailure(exc);
     }
 
     private void finishWithIndexingFailure(Exception exc) {
         stats.incrementIndexingFailures();
-        onFailure(exc);
-        doSaveState(finishAndSetState(), position.get(), this::afterFinishOrFailure);
+        finishWithFailure(exc);
     }
 
     private void finishWithFailure(Exception exc) {
         onFailure(exc);
-        finishAndSetState();
-        afterFinishOrFailure();
+        doSaveState(finishAndSetState(), position.get(), this::afterFinishOrFailure);
     }
 
     private IndexerState finishAndSetState() {

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexerTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexerTests.java
@@ -357,7 +357,6 @@ public class AsyncTwoPhaseIndexerTests extends ESTestCase {
 
         @Override
         protected void doSaveState(IndexerState state, Integer position, Runnable next) {
-            fail("should not be called");
         }
 
         @Override

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexerTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/indexing/AsyncTwoPhaseIndexerTests.java
@@ -356,8 +356,7 @@ public class AsyncTwoPhaseIndexerTests extends ESTestCase {
         }
 
         @Override
-        protected void doSaveState(IndexerState state, Integer position, Runnable next) {
-        }
+        protected void doSaveState(IndexerState state, Integer position, Runnable next) {}
 
         @Override
         protected void onFailure(Exception exc) {

